### DISCUSSION TITLE
Is it possible to have certain UniTasks run in a single non-Unity thread

### DISCUSSION THREAD

**Discussion by dimozaprianov** [Cysharp/UniTask#360] [Q&A]
Hi, I'm working on a turn based game where the game logic is running in non-Unity thread. It is this way so that it can be easily extracted and ran on a server. Is it possible to use UniTasks and have them executed in that same thread so that I don't have to handle multithreading issues?

Thanks!

**Comment by neuecc** [comment]
Since UniTask is tightly coupled to PlayerLoop,
getting it back to a single thread may not be easy. (What state is the thread in while it is awaiting in the first place?)

Yes, I would probably implement `SwitchToYourThread` with an implementation like `SwitchToThreadPool`.
